ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -200,6 +201,20 @@ public Optional<UInt256> storageAt(
         address, blockHash, account -> account.getStorageValue(storageIndex), UInt256.ZERO);
   }
 
+  /**
+   * Returns the storage root hash for the given address.
+   *
+   * @param address The address of the account that owns the storage being queried.
+   * @param blockNumber The blockNumber that is being queried.
+   * @return The storage root hash for the account being queried.
+   */
+  public Optional<Hash> storageRoot(final Address address, final long blockNumber) {
+    final Hash blockHash =
```

Comment:
why in case the block is not found we do not just return empty, instead of searching for the `Hash.EMPTY` block? What does a search with block `Hash.EMPTY` return?